lib/poller.js

```diff
@@ -187,21 +189,42 @@ class Poller {
    * @param {Object} descriptor secret descriptor
    */
   _isPermitted (namespace, descriptor) {
-    const role = descriptor.roleArn
     let allowed = true
     let reason = ''
 
-    if (!namespace.metadata.annotations || !role) {
+    if (!namespace.metadata.annotations) {
+      return {
+        allowed, reason
+      }
+    }
+
+    const externalData = descriptor.data || descriptor.properties
+    const namingConvention = namespace.metadata.annotations[this._namingPermittedAnnotation]
+
+    if (namingConvention) {
+      externalData.forEach((secretProperty, index) => {
+        const reNaming = new RegExp(namingConvention)
+        if (!reNaming.test(secretProperty.key)) {
+          allowed = false
+          reason = `key name does not match naming convention ${namingConvention}`
```

Comment:
might as well return here, now we keep looping and the error message will be for the last key, as opposed to early exit with error for the first key name

Reply:
Yep you are right. I fixed this. Thanks!